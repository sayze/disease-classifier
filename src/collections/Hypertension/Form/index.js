import React from 'react'
import { Formik } from 'formik'
import * as Yup from 'yup'
import { Button, TextField } from '@material-ui/core'
import PropTypes from 'prop-types'
import { makeStyles } from '@material-ui/core/styles'
import Datepicker from 'components/Datepicker'

const useStyles = makeStyles(theme => ({
  inputField: {
    marginBottom: theme.spacing(3),
  },
}))

const Form = ({ onFormSubmit }) => {
  const classes = useStyles()
  return (
    <Formik
      initialValues={{ dialBp: 0, sysBp: 0, date: new Date() }}
      validationSchema={Yup.object({
        sysBp: Yup.number()
          .typeError('Systolic Blood Pressure must be a number')
          .required('Systolic Blood Pressure is required'),
        dialBp: Yup.number()
          .typeError('Diastolic Blood Pressure must be a number')
          .required('Diastolic Blood Pressure is required'),
      })}
      onSubmit={onFormSubmit}
    >
      {({ errors, touched, values, handleChange, handleSubmit, setFieldValue }) => (
        <>
          <Datepicker
            fullWidth
            autoFocus
            name="date"
            onChange={date => setFieldValue('date', date)}
            className={classes.inputField}
            value={values.date}
            label="Date of Reading"
          />
          <TextField
            fullWidth
            className={classes.inputField}
            name="sysBp"
            variant="outlined"
            label="Systolic Blood Pressure"
            error={Boolean(touched.sysBp && errors.sysBp)}
            helperText={errors.sysBp && touched.sysBp ? errors.sysBp : 'Number for Systolic Blood Pressure'}
            onChange={handleChange}
            defaultValue={values.sysBp}
          />
          <TextField
            fullWidth
            className={classes.inputField}
            name="dialBp"
            variant="outlined"
            label="Diastolic Blood Pressure"
            error={Boolean(touched.dialBp && errors.dialBp)}
            helperText={errors.dialBp && touched.dialBp ? errors.dialBp : 'Number for Diastolic Blood Pressure'}
            onChange={handleChange}
            defaultValue={values.dialBp}
          />
          <Button variant="contained" onClick={handleSubmit} color="primary" fullWidth>
            Classify
          </Button>
        </>
      )}
    </Formik>
  )
}

Form.defaultProps = {
  sysBp: 0,
  dialBp: 0,
  onFormSubmit: () => {},
}

Form.propTypes = {
  sysBp: PropTypes.number,
  dialBp: PropTypes.number,
  onFormSubmit: PropTypes.func,
}

export default Form
