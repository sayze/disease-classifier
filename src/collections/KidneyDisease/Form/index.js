import React from 'react'
import { Formik } from 'formik'
import * as Yup from 'yup'
import { Button, TextField } from '@material-ui/core'
import PropTypes from 'prop-types'
import { makeStyles } from '@material-ui/core/styles'

const useStyles = makeStyles(theme => ({
  inputField: {
    marginBottom: theme.spacing(3),
  },
}))

const Form = ({ onFormSubmit }) => {
  const classes = useStyles()
  return (
    <Formik
      initialValues={{ eGFR: 0 }}
      validationSchema={Yup.object({
        eGFR: Yup.number()
          .typeError('eGFR must be a number')
          .required('eGFR is required'),
      })}
      onSubmit={onFormSubmit}
    >
      {({ errors, touched, values, handleChange, handleSubmit }) => (
        <>
          <TextField
            fullWidth
            autoFocus
            className={classes.inputField}
            name="eGFR"
            variant="outlined"
            label="eGFR"
            error={Boolean(touched.eGFR && errors.eGFR)}
            helperText={errors.eGFR && touched.eGFR ? errors.eGFR : 'Number for eGFR'}
            onChange={handleChange}
            defaultValue={values.eGFR}
          />
          <Button variant="contained" onClick={handleSubmit} color="primary" fullWidth>
            Classify
          </Button>
        </>
      )}
    </Formik>
  )
}

Form.defaultProps = {
  eGFR: 0,
  onFormSubmit: () => {},
}

Form.propTypes = {
  eGFR: PropTypes.number,
}

export default Form
