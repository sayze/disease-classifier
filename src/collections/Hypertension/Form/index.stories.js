import React from 'react'
import Form from '.'

export default {
  title: 'Hypertension Form',
  component: Form,
}

export const Default = () => <Form />
