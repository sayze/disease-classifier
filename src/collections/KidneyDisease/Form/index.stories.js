import React from 'react'
import Form from '.'

export default {
  title: 'Kidney Disease Form',
  component: Form,
}

export const Default = () => <Form />
