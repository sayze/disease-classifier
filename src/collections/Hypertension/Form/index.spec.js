import React from 'react'
import { shallow } from 'enzyme'
import Form from '.'

describe('Hypertension Form', () => {
  it('should render a default hypertension form', () => {
    const component = shallow(<Form />)
    expect(component).toMatchSnapshot()
  })
})
