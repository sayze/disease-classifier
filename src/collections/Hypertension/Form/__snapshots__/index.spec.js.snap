// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hypertension Form should render a default hypertension form 1`] = `
<Formik
  enableReinitialize={false}
  initialValues={
    Object {
      "dialBp": 0,
      "sysBp": 0,
    }
  }
  isInitialValid={false}
  onSubmit={[Function]}
  validateOnBlur={true}
  validateOnChange={true}
  validationSchema={
    ObjectSchema {
      "_blacklist": RefSet {
        "list": Set {},
        "refs": Map {},
      },
      "_conditions": Array [],
      "_defaultDefault": [Function],
      "_deps": Array [],
      "_excludedEdges": Array [],
      "_exclusive": Object {},
      "_mutate": undefined,
      "_nodes": Array [
        "dialBp",
        "sysBp",
      ],
      "_options": Object {
        "abortEarly": true,
        "recursive": true,
      },
      "_type": "object",
      "_typeError": [Function],
      "_whitelist": RefSet {
        "list": Set {},
        "refs": Map {},
      },
      "fields": Object {
        "dialBp": NumberSchema {
          "_blacklist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "_conditions": Array [],
          "_deps": Array [],
          "_exclusive": Object {
            "required": true,
          },
          "_mutate": undefined,
          "_options": Object {
            "abortEarly": true,
            "recursive": true,
          },
          "_type": "number",
          "_typeError": [Function],
          "_whitelist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "tests": Array [
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
        },
        "sysBp": NumberSchema {
          "_blacklist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "_conditions": Array [],
          "_deps": Array [],
          "_exclusive": Object {
            "required": true,
          },
          "_mutate": undefined,
          "_options": Object {
            "abortEarly": true,
            "recursive": true,
          },
          "_type": "number",
          "_typeError": [Function],
          "_whitelist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "tests": Array [
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
        },
      },
      "tests": Array [],
      "transforms": Array [
        [Function],
      ],
    }
  }
>
  <Component />
</Formik>
`;
